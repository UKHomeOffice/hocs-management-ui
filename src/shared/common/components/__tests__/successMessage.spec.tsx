import React from 'react';
import { render } from '@testing-library/react';

import SuccessMessage from '../successMessage';
import { Location } from 'history';

const mockLocation: Location = {
    state: {
        successMessage: '__successMessage__'
    },
    pathname: '',
    search: '',
    hash: ''
};

describe('when the SuccessMessage is rendered', () => {
    it('should show the success message', () => {
        expect(
            render(<SuccessMessage location={mockLocation} />)
        ).toMatchSnapshot();
    });
    it('should not display when there is not a message', () => {
        mockLocation.state = undefined;
        expect(
            render(<SuccessMessage location={mockLocation} />)
        ).toMatchSnapshot();
    });
});
