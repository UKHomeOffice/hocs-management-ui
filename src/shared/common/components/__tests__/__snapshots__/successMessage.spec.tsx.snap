// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the SuccessMessage is rendered should not display when there is not a message 1`] = `null`;

exports[`when the SuccessMessage is rendered should show the success message 1`] = `
<div
  class="panel panel-border-wide alert-success"
>
  <h2
    class="govuk-heading-s"
  >
    Success
  </h2>
  <p
    class="govuk-body"
  >
    __successMessage__
  </p>
</div>
`;
